base/util.jl

```diff
@@ -574,3 +574,62 @@ function runtests(tests = ["all"]; ncores::Int = ceil(Int, Sys.CPU_THREADS::Int
               "including error messages above and the output of versioninfo():\n$(read(buf, String))")
     end
 end
+
+function maybe_resolve_global(m::Module, name::Symbol)
+    try
+        return Some(getfield(m, name))
+    catch
+        return nothing
+    end
+end
+function maybe_resolve_global(m::Module, ex)
+    Meta.isexpr(ex, :., 2) || return nothing
+    a, b = ex.args
+    a isa Union{Symbol,Expr} || return nothing
+    b isa QuoteNode || return nothing
+    c = b.value
+    c isa Symbol || return nothing
+    obj = @something(maybe_resolve_global(m, a), return nothing)
+    if obj isa Module
+        return  maybe_resolve_global(obj, c)
+    else
+        return nothing
+    end
+end
+
+"""
+    inbounds_warning(__source__::LineNumberNode, __module__::Module, blk::Expr)
+
+Called from `@inbounds` macro to warn some known pitfalls.
+"""
+function inbounds_warning(__source__::LineNumberNode, __module__::Module, blk::Expr)
+    hint = """
+    HINT: It is recommended to avoid using `@inbounds` around the code you don't write.
+    A macro can introduce arbitrary code which may not satisfy the requirement of
+    `@inbounds`.
+    """
```

Comment:
This might be too opinionated so I'm happy to remove :)

Reply:
Yeah I would remove this